registry/storage/vacuum.go

```diff
@@ -84,6 +84,38 @@ func (v Vacuum) RemoveManifest(name string, dgst digest.Digest, tags []string) e
 	return v.driver.Delete(v.ctx, manifestPath)
 }
 
+// RemoveLayerLink removes a layer link from the filesystem
+func (v Vacuum) RemoveLayerLink(manifestName, dgst string) error {
+	d, err := digest.Parse(dgst)
+	if err != nil {
+		return err
+	}
+
+	layerLinkPath, err := pathFor(layerLinkPathSpec{name: manifestName, digest: d})
+	if err != nil {
+		return err
+	}
+
+	dcontext.GetLogger(v.ctx).Infof("Deleting layer link path : %s", layerLinkPath)
+
+	_, err = v.driver.Stat(v.ctx, layerLinkPath)
+	if err != nil {
+		switch err := err.(type) {
+		case driver.PathNotFoundError:
+			return nil
+		default:
+			return err
+		}
+	}
+
+	err = v.driver.Delete(v.ctx, layerLinkPath)
```

Comment:
Nit: Could `return v.driver.Delete(...)` since we're not doing anything with the error other than returning it.